array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,24 @@
+/* exported chunk */
+function chunk(array, size) {
+  const arrayCopy = array;
+  let currentChunk = [];
+  const finalChunk = [];
+
+  for (let i = 0; i < arrayCopy.length; i++) {
+    if (currentChunk.length < size) {
+      currentChunk.push(arrayCopy[i]);
+    } else if (i === size) {
+      finalChunk.push(currentChunk);
+      currentChunk = [];
+      currentChunk.push(arrayCopy[i]);
+    } else if (i % size === 0) {
+      finalChunk.push(currentChunk);
+      currentChunk = [];
+      currentChunk.push(arrayCopy[i]);
+    }
```

Comment:
since these two code blocks are the same, you can combine them

```js
    } else if (i === size || i % size === 0) {
      finalChunk.push(currentChunk);
      currentChunk = [];
      currentChunk.push(arrayCopy[i]);
    } 
```